python/sglang/srt/utils.py

```diff
@@ -1543,6 +1549,7 @@ def init_custom_process_group(
         group_name=group_name,
         **{pg_options_param_name: pg_options},
         timeout=timeout,
+        device_id=device_idx,
```

Comment:
I've just tried this PR with slime. It seems that if I remove the device_id relevant logic, I can run with dp attention on, while it will stuck with the `device_id`. Could you give some more details about the following reason?
> When creating an NCCL communication group, failing to specify the device ID for the newly created weight communication group can cause an NCCL error indicating a duplicate GPU.